agent/ui_endpoint.go

```diff
@@ -200,17 +251,56 @@ RPC:
 		return nil, err
 	}
 
-	return summarizeServices(out.Dump, nil, s.agent.config, args.Datacenter), nil
+	summaries, _ := summarizeServices(out.Dump, s.agent.config, args.Datacenter)
+	return prepSummaryOutput(summaries, false), nil
 }
 
-// TODO (freddy): Refactor to split up for the two use cases
-func summarizeServices(dump structs.ServiceDump, gateways structs.GatewayServices, cfg *config.RuntimeConfig, dc string) []*ServiceSummary {
-	// Collect the summary information
-	var services []structs.ServiceName
-	summary := make(map[structs.ServiceName]*ServiceSummary)
+func (s *HTTPHandlers) UIServiceTopology(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
```

Comment:
I find a comment with the API URL path this is mounted on is super useful when browsing around these. We're not super consistent but most HTTP handler endpoints do have those.